sycl/source/detail/device_global_map_entry.cpp

```diff
@@ -67,14 +67,19 @@ DeviceGlobalUSMMem &DeviceGlobalMapEntry::getOrAllocateDeviceGlobalUSM(
          "USM allocation for device and context already happened.");
   DeviceGlobalUSMMem &NewAlloc = NewAllocIt.first->second;
 
-  // Zero-initialize here and save the event.
+  // Initialize here and save the event.
   {
-    std::lock_guard<std::mutex> Lock(NewAlloc.MZeroInitEventMutex);
+    std::lock_guard<std::mutex> Lock(NewAlloc.MInitEventMutex);
     sycl::detail::pi::PiEvent InitEvent;
-    MemoryManager::fill_usm(NewAlloc.MPtr, QueueImpl, MDeviceGlobalTSize, 0,
+    // C++ guarantees members appear in memory in the order they are declared,
+    // so since the member variable that contains the initial contents of the
+    // device_global is right after the usm_ptr member variable we can do
+    // some pointer arithmetic to memcopy over this value to the usm_ptr.
+    MemoryManager::copy_usm(MDeviceGlobalPtr + sizeof(MDeviceGlobalPtr),
```

Comment:
Nit: I would prefer that we use `sizeof(uintptr_t)` to make it a bit clearer here that what we are skipping is a single pointer, not actually related to `MDeviceGlobalPtr`'s size.

Reply:
Updated!

Reply:
So it turns out that we won't be able to pass the CI using `sizeof(uintptr_t)` as it produces a compiler warning and since we are treating all warnings as errors, we fail. I'm changing this back to `MDeviceGlobalPtr` and adding an additional comment explaining the rationale.